File: questions/leetcode/solution_0001/python/Solution.py
#!/anaconda3/bin/python
# @Time    : 2021/4/26 11:37 下午
# @Author  : warbie
# @File    : Solution
# @Software: PyCharm
# @Description:

from typing import List, Dict


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_dict: Dict[int, int] = dict()
        for i, num in enumerate(nums):
            if target - num in num_dict:
                return [num_dict[target - num], i]
            num_dict[num] = i
        return []

    def checkDuplicated(self, inputStr: str) -> bool:
        values: List[str] = str.split(inputStr, ',')
        str_dict: Dict[str, int] = dict()
        for v in values:
            if v in str_dict:
                return True
            str_dict[v] = 1
        return False



if __name__ == '__main__':
    solution = Solution()
    print(solution.twoSum([2, 7, 11, 15], 9))
    print(solution.checkDuplicated("1,2,3,4"))

File: questions/leetcode/solution_0001/java/Solution.java
package solution_0001.java;

import java.util.HashMap;
import java.util.Map;

/**
 * 两数之和
 *
 * @author ZhouYi
 * 2021/3/28 11:59 下午
 */
class Solution {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> numMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (numMap.containsKey(nums[i])) {
                return new int[]{numMap.get(nums[i]), i};
            }
            numMap.put(target - nums[i], i);
        }

        return null;
    }
}

class Main {
    public static void main(String[] main) {
        System.out.println(new Solution().twoSum(new int[]{2, 3, 4}, 5));
    }
}
